solr/solrj/src/java/org/apache/solr/common/util/PathTrie.java

```diff
@@ -169,23 +173,45 @@ public T lookup(List<String> pieces, int i, Map<String, String> templateValues)
     }
 
     /**
-     *
-     * @param pathSegments pieces in the url /a/b/c has pieces as 'a' , 'b' , 'c'
-     * @param index current index of the pieces that we are looking at in /a/b/c 0='a' and 1='b'
+     * @param pathSegments      pieces in the url /a/b/c has pieces as 'a' , 'b' , 'c'
+     * @param index             current index of the pieces that we are looking at in /a/b/c 0='a' and 1='b'
      * @param templateVariables The mapping of template variable to its value
      * @param availableSubPaths If not null , available sub paths will be returned in this set
      */
     public T lookup(List<String> pathSegments, int index, Map<String, String> templateVariables, Set<String> availableSubPaths) {
       if (templateName != null) templateVariables.put(templateName, pathSegments.get(index - 1));
       if (pathSegments.size() < index + 1) {
         findAvailableChildren("", availableSubPaths);
+        if(obj == null ){//this is not a leaf node
```

Comment:
There should be space after 'if' and before the '('. @markrmiller used to be very particular about it, and I learnt it from him.